### ISSUE TITLE
Move AEM keystore password secret to parameter store in aws-resources

### ISSUE THREAD

**Issue by cliffano** [shinesolutions/packer-aem#124]
AEM keystore password is currently configurable in YAML inventory file https://github.com/shinesolutions/packer-aem/blob/master/conf/ansible/inventory/group_vars/defaults.yaml#L61 , this needs to be moved to AWS Systems Manager parameter store similar to AEM license.

Please also update the upgrade guide https://github.com/shinesolutions/packer-aem/blob/master/docs/upgrade-guide.md 

And review hello world config's Packer AEM profiles, whether there's any specified keystore password there https://github.com/shinesolutions/aem-helloworld-config/tree/master/packer-aem . If any does exist, it has to be removed.

**Comment by phillipi-shinesolutions** [comment]
Hi Cliff,

I assume you would like me to remove the keystore value from the ansible conf once it has been moved?

https://github.com/shinesolutions/packer-aem/blob/master/conf/ansible/inventory/group_vars/defaults.yaml#L61

**Comment by cliffano** [comment]
Yes correct.

**Comment by cliffano** [comment]
Done. To be included in version 3.8.0 .

**Comment by cliffano** [comment]
Reopening this issue due to keystore password consumption still relies on user config.

**Comment by cliffano** [comment]
To complete this ticket, the AMI baking should retrieve the keystore password from parameter store instead of relying on YAML config.

Have a look at this example on how to retrieve a parameter store object https://github.com/shinesolutions/packer-aem/blob/master/provisioners/puppet/modules/config/manifests/license.pp#L60 . Unfortunately there is no built-in Puppet module for dealing with parameter store, hence we have to resort to CLI.

What needs to be done:
- Remove default keystore password `aem.keystore_password` from Ansible default YAML
- Remove configuration documentation of `aem.keystore_password`
- Remoev the usage of that keystore password variable in `templates/ansible/hieradata.j2` 
  This means we'll no longer inject `aem_curator::install_author::aem_keystore_password` and `aem_curator::install_publish::aem_keystore_password`
- Instead, we'll be passing the value via Puppet manifests at `provisioners/puppet/manifests/author.pp` and `provisioners/puppet/manifests/publish.pp`
Inside author.pp and publish.pp, you'll find `include aem_curator::install_author` and `include aem_curator::install_publish` respectively, so instead of calling those manifests using `define`, you need to change them to use
```
aem_curator::installauthor { 'Install AEM Author':
  aem_keystore_password => passthevalueretreivedfromawscli,
}
```
The `aem_keystore_password` itself is defined https://github.com/shinesolutions/puppet-aem-curator/blob/master/manifests/install_author.pp#L35 .

The same applies for publish.

**Comment by cliffano** [comment]
Done. To be included in release 4.0.0 .
